CRM/Contribute/Form/Search.php

```diff
@@ -163,6 +153,18 @@ public function setDefaultValues() {
         'Completed'
       );
     }
+
+    // The membership or contribution id could be set on the form if viewing
+    // an embedded block on ParticipantView or MembershipView.
+    $memberShipId = CRM_Utils_Request::retrieve('memberId', 'Positive', $this);
+    if (isset($memberShipId)) {
+      $this->_defaults['contribution_membership_id'] = $memberShipId;
```

Comment:
Maybe fix the weird capitalisation of $memberShipId that was in the original?  No boats here ...